### DISCUSSION TITLE
PublishDir flag doesn't work when publishing Mac Catalyst project

### DISCUSSION THREAD

**Discussion by Juriyx** [unoplatform/uno#16873] [Q&A]
Hello. After a command to build a single Uno project for all platforms `msbuild /r /p:Configuration=$(BuildConfiguration)` I use this command to publish one of the platforms - Mac Catalyst:

`msbuild /t:Publish /p:Configuration=Release /p:Platform=AnyCPU /p:TargetFramework=net8.0-maccatalyst /p:PublishDir="$(Build.ArtifactStagingDirectory)/Mac Catalyst"`

PublishDir flag works successfully on other platforms, but not on Mac Catalyst - the application package is not created and moved to $(Build.ArtifactStagingDirectory)

**Comment by agneszitte** [comment]
cc @jeromelaban, @spouliot

**Reply by jeromelaban** [reply]
@Juriyx this is expected, catalyst projects cannot be built on Windows. You'll need to build on a mac.

**Reply by Juriyx** [reply]
> @Juriyx this is expected, catalyst projects cannot be built on Windows. You'll need to build on a mac.

Ok, thanks. Is it possible to exclude iOS and Mac Catalyst when building?

**Reply by jeromelaban** [reply]
Can you specify in which context you are having an issue that would need ios and catalyst to be removed?

**Reply by Juriyx** [reply]
When we publish projects on Windows, we wouldn't want to waste time building them since they would have to be built on macOS anyway, as you correctly wrote

**Reply by jeromelaban** [reply]
Thanks, that makes sense.

At this time, msbuild does not support explicitly building for a set of targets. Setting the `TargetFrameworks` property to limit the number of targets, while possible in some cases, may break class libraries. You can build a single TFM by using the `-f`  (with `dotnet build`) or `/p:TargetFramework=` (with `msbuild`) to build for a single target framework.

For the original MacCatalyst issue, this part is handled by the .NET SDK itself, and we may need to add a workaround to move the file at the end. In the meantime, you can move the file in a custom `Target` in your csproj.
